src/main/java/magentoPage/ProductsPage.java

```diff
@@ -0,0 +1,73 @@
+package magentoPage;
+
+import object.product.ProductWishlist;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import utils.property.PropertyUtil;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Properties;
+
+public class ProductsPage {
+    private WebDriver driver;
+    private int random;
+    private By gridProducts = By.xpath("//div[contains(@class,'grid products')]//li");
+    private final Properties testData = PropertyUtil.getProperties("testsData.properties");
+
+    public ProductsPage(WebDriver driver) {
+        this.driver = driver;
+        PageFactory.initElements(driver, this);
```

Comment:
Do you need it there is you don't have locators with @FindBy annotation ?

Reply:
fixed